src/EditorFeatures/Core.Wpf/InlineErrors/InlineErrorTaggerProvider.cs

```diff
@@ -0,0 +1,124 @@
+{
+    [Export(typeof(ITaggerProvider))]
+    [ContentType(ContentTypeNames.RoslynContentType)]
+    [ContentType(ContentTypeNames.XamlContentType)]
+    [TagType(typeof(InlineErrorTag))]
+    internal class InlineErrorTaggerProvider : AbstractDiagnosticsAdornmentTaggerProvider<InlineErrorTag>
+    {
+        private readonly IEditorFormatMap _editorFormatMap;
+        protected override IEnumerable<PerLanguageOption2<bool>> PerLanguageOptions => SpecializedCollections.SingletonEnumerable(FeatureOnOffOptions.InlineErrors);
+
+        protected internal override bool IsEnabled => true;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public InlineErrorTaggerProvider(
+            IThreadingContext threadingContext,
+            IEditorFormatMapService editorFormatMapService,
+            IDiagnosticService diagnosticService,
+            IAsynchronousOperationListenerProvider listenerProvider)
+            : base(threadingContext, diagnosticService, listenerProvider)
+        {
+            _editorFormatMap = editorFormatMapService.GetEditorFormatMap("text");
+        }
+
+        protected override SnapshotSpan AdjustSnapshotSpan(SnapshotSpan span, int minimumLength)
+        => AdjustSnapshotSpan(span, minimumLength, int.MaxValue);
+
+        protected static new SnapshotSpan AdjustSnapshotSpan(SnapshotSpan span, int minimumLength, int maximumLength)
+        {
+            var snapshot = span.Snapshot;
+
+            // new length
+            var length = Math.Min(Math.Max(span.Length, minimumLength), maximumLength);
+
+            // make sure start + length is smaller than snapshot.Length and start is >= 0
+            var start = Math.Max(0, Math.Min(span.Start, snapshot.Length - length));
+
+            // make sure length is smaller than snapshot.Length which can happen if start == 0
+            return new SnapshotSpan(snapshot, start, Math.Min(start + length, snapshot.Length) - start);
+        }
+
+        protected internal override bool IncludeDiagnostic(DiagnosticData diagnostic)
+        {
+            return
+                (diagnostic.Severity == DiagnosticSeverity.Warning || diagnostic.Severity == DiagnosticSeverity.Error) &&
+                !string.IsNullOrWhiteSpace(diagnostic.Message);
+        }
+
+        /// <summary>
+        /// Creates the InlineErrorTag with the error distinction
+        /// </summary>
+        protected override InlineErrorTag? CreateTag(Workspace workspace, DiagnosticData diagnostic)
+        {
+            Debug.Assert(!string.IsNullOrWhiteSpace(diagnostic.Message));
+            var errorType = GetErrorTypeFromDiagnostic(diagnostic);
+            if (errorType is null)
+            {
+                return null;
+            }
+
+            return new InlineErrorTag(errorType, diagnostic, _editorFormatMap);
+        }
+
+        private static string? GetErrorTypeFromDiagnostic(DiagnosticData diagnostic)
```

Comment:
why not check this in IncludeDiagnostic? i don't get the point of including it just to not makea  tag for it later.

Reply:
I think the line you were referring to here was the if statement for the diagnostic suppression? Moved to IncludeDiagnostic